lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -370,12 +370,43 @@ struct EliminateUnaryMuxesPattern : OpRewritePattern<MuxOp> {
   }
 };
 
+struct EliminateCBranchIntoMuxPattern : OpRewritePattern<MuxOp> {
+  using mlir::OpRewritePattern<MuxOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(MuxOp op,
+                                PatternRewriter &rewriter) const override {
+
+    if (op.getDataOperands().size() != 2)
```

Comment:
Nit: store `op.getDataOperands()` in a variable to avoid calling it 3 times in this function.

Reply:
Will do.